### PR TITLE
8270961: [TESTBUG] Move GotWrongOOMEException into vm.share.gc package

### PR DESCRIPTION
**Pull request by lmesnik**
The small refactoring which consolidates shared code in the shared library. Make project more IDE-friendly.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8270961](https://bugs.openjdk.java.net/browse/JDK-8270961): [TESTBUG] Move GotWrongOOMEException into vm.share.gc package

### Reviewers
 * [Kim Barrett](https://openjdk.java.net/census#kbarrett) (@kimbarrett - **Reviewer**) ⚠️ Review applies to 990bff2f0ab70fe1a0aaa3a1bb6fd3b5f9523258
 * [Thomas Schatzl](https://openjdk.java.net/census#tschatzl) (@tschatzl - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/4844/head:pull/4844` \
`$ git checkout pull/4844`

Update a local copy of the PR: \
`$ git checkout pull/4844` \
`$ git pull https://git.openjdk.java.net/jdk pull/4844/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 4844`

View PR using the GUI difftool: \
`$ git pr show -t 4844`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/4844.diff">https://git.openjdk.java.net/jdk/pull/4844.diff</a>

</details>

### CODE REVIEW

File: test/hotspot/jtreg/vmTestbase/metaspace/stressHierarchy/common/StressHierarchyBaseClass.java [hunk 1]
```diff
@@ -109,8 +109,13 @@ public void run() {
...
             System.out.println("Whole test took " + ((System.currentTimeMillis() - startTimeStamp)/1000/60.0) +" min");
             log.info("Test PASSED");
-        } catch (GotWrongOOMEException e) {
-            log.info("GotWrongOOMEExc: " + e.getMessage());
+        } catch (HeapOOMEException e) {
+            /**
+             * Usually this means that we got OOME:heap while trying to gain OOME:metaspace.
+             * We pass test in this case as this breaks test logic. We have dedicated test configurations
+             * for OOME:heap provoking class unloading, that why we are not missing test coverage here.
+             */
```
Comment (kimbarrett) [inline test/hotspot/jtreg/vmTestbase/metaspace/stressHierarchy/common/StressHierarchyBaseClass.java:117]: This comment is unnecessary, as it adds nothing over the log message that follows.

File: test/hotspot/jtreg/vmTestbase/vm/share/gc/HeapOOMEException.java [hunk 2]
```diff
@@ -20,18 +20,16 @@
...
  * questions.
  */
-package metaspace.stressHierarchy.common.exceptions;
+package vm.share.gc;
 
 /**
- * Usually this means that we got OOME:heap while trying to gain OOME:metaspace.
- * We pass test in this case as this breaks test logic. We have dedicated test configurations
- * for OOME:heap provoking class unloading, that why we are not missing test coverage here.
+ * This class is used to differ OOME in metaspace and heap when trigger class unloading.
```
Comment (kimbarrett) [inline test/hotspot/jtreg/vmTestbase/vm/share/gc/HeapOOMEException.java:26]: That doesn't scan so well.  Perhaps something like
"This class is used to distinguish between OOME in metaspace and OOME in heap when triggering class unloading."

**Review by kimbarrett — APPROVED** [review]
Some minor commenting issues, but otherwise looks good.

File: test/hotspot/jtreg/vmTestbase/metaspace/stressHierarchy/common/StressHierarchyBaseClass.java (hunk 1 above)
Comment (lmesnik) [reply test/hotspot/jtreg/vmTestbase/metaspace/stressHierarchy/common/StressHierarchyBaseClass.java]: Thanks! I just moved it and missed that is the same as the log message.

**Review by tschatzl — APPROVED** [review]

**Pull request closed without merging**